orchagent/portsorch.cpp

```diff
@@ -4186,6 +4186,25 @@ void PortsOrch::doPortTask(Consumer &consumer)
                         it = m_portListLaneMap.erase(it);
                         continue;
                     }
+                    else
+                    {
+                        auto pCfg = m_lanesAliasSpeedMap[it->first];
+                        sai_uint32_t speed = 0;
+                        
+                        getPortSpeed(it->second, speed);
+                        if (pCfg.speed.is_set && speed != pCfg.speed.value)
+                        {
+                            SWSS_LOG_DEBUG("Port %" PRIx64 " speed mismatch: cfg speed %u, hw speed %u lane: %s",
+                                    it->second, 
+                                    pCfg.speed.value, 
+                                    speed,
+                                    swss::join(" ", it->first.cbegin(), it->first.cend()).c_str()
+                            );
+                            portsToRemoveList.push_back(it->second);
+                            it = m_portListLaneMap.erase(it);
+                            continue;
+                        }
+                    }
```

Comment:
@prgeor port recreation due to invalid speed looks like an overhead. What's wrong with the speed configuration which takes place after port initialization? IMHO this looks like a W/A